app/assets/javascripts/controllers/current_deploys_badge_ctrl.js

```diff
@@ -1,33 +1,30 @@
 samson.controller('currentDeployBadgeCtrl', function($scope, $http, SseFactory) {
   'use strict';
 
-  $scope.currentActiveDeploys = 0;
+  var $badge = $('#currentDeploysBadge');
+  $scope.currentActiveDeploys = parseInt($badge.data('count'), 10);
 
-  SseFactory.on('deploys', function(msg) {
-    if (msg.type === 'new') {
-      $scope.currentActiveDeploys += 1;
-    } else if (msg.type === 'finish') {
-      $scope.currentActiveDeploys -= 1;
-    }
-    updateBadge();
-  });
+  // connect to stream for updates once a user has the page open for a while
+  setTimeout(function(){
+    SseFactory.on('deploys', function(msg) {
+      if (msg.type === 'new') {
+        $scope.currentActiveDeploys += 1;
+      } else if (msg.type === 'finish') {
+        $scope.currentActiveDeploys -= 1;
+      }
+      updateBadge();
+    });
+  }, 5000);
 
   function updateBadge() {
     _.defer(function() { $scope.$apply(); });
     if ($scope.currentActiveDeploys > 0) {
-      $('#currentDeploysBadge').show();
+      $badge.show();
     } else {
       $scope.currentActiveDeploys = 0;
-      $('#currentDeploysBadge').hide();
+      $badge.hide();
     }
   }
 
-  function init() {
-    $http.get('/deploys/active_count.json').success(function(result) {
```

Comment:
You will still need this call otherwise the badge count will be potentially incorrect since it'll miss all the starting/stopping deploy events during the 5 seconds.